src/SonarQube.Client/Api/DefaultConfiguration.cs

```diff
@@ -0,0 +1,50 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarQube.Client.Api.Requests
+{
+    public static class DefaultConfiguration
+    {
+        public static RequestFactory Configure(RequestFactory requestFactory)
+        {
+            requestFactory
+                .RegisterRequest<IGetPluginsRequest, V2_10.GetPluginsRequest>("2.1")
```

Comment:
Perhaps the version of SQ should be a property on the Request object (rather than being a separate parameter that is passed n RegisterRequest).
The request is version-specific so it has to know the SQ version number implicitly - there doesn't seem to be any advantage in having the caller to requestFactory know this too.

Reply:
Warning: this cannot be an instance property like `Path` because in the factory we don't have instances before resolve the versioned request and that's too late.

Reply:
I could invert the code, for example in DefaultConfiguration:
```
```
GetQualiryProfilesRequest.Register(factory);
...
```

Bug imo this does not improve the code that much as we cannot use a base class to define the method and we will have to duplicate...

Reply:
It breaks encapsulation and information hiding, but since we're using static registration I guess that risk of us getting it wrong is low. I'm ok with leaving it as it is.

FYI I was trying to think of an alternative approach, and the only one I could come up with is to create a request factory interface for each type of request. That would mean that current _requestFactory_ wouldn't have to know the versions that are supported by each request type, and it would also mean each factory interface could have a strongly-typed _Create_ method with the parameters required for that API (instead of the current _configure_ lambda).